### PR TITLE
SI-6937 core type tags are no longer referentially unique

### PR DESCRIPTION
**Pull request by xeno-by**
Type tag factory used to evaluate the provided type creator in the
context of the initial mirror in order to maintain referential equality
of instances of standard tags. Unfortunately this evaluation might fail
if the mirror provided doesn't contain the classes being referred to.
Therefore I think we should avoid evaluating type creators there.

Note that failure of evaluation doesn't mean that there's something
bad going on. When one creates a type tag, the correct mirror /
classloader to interpret that tag in might be unknown (like it happens
here). This is okay, and this is exactly what the 2.10.0-M4 refactoring
has addressed.

Something like `res2.typeTag[A].in(currentMirror)` should be okay.

### CODE REVIEW

**Comment by xeno-by** [comment]
review @retronym @rkuhn

File: test/files/run/t6937.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import scala.tools.partest.ReplTest
+
+object Test extends ReplTest {
+  def code = """
+    import scala.reflect.runtime.{universe => ru}
+    import scala.reflect.runtime.{currentMirror => cm}
+    import scala.reflect.api.{Universe => ApiUniverse}
+    class A
+    lazy val apiru = ru: ApiUniverse
```
Comment (rkuhn) [inline test/files/run/t6937.scala:9]: why lazy?

**Comment by rkuhn** [comment]
I’m not qualified to comment on referential equality of core type tags, but if this fixes my use-case then I’m happy ;-)

File: test/files/run/t6937.scala (hunk 1 above)
Comment (xeno-by) [inline test/files/run/t6937.scala:9]: so that its value doesn't get printed by the repl (ru.toString is smth like `JavaUniverse@5a4ee78a`)

Comment (rkuhn) [inline test/files/run/t6937.scala:9]: ah okay, as long as the `lazy` is not needed for correct function of the `typeTag` then everything is fine

Comment (xeno-by) [inline test/files/run/t6937.scala:9]: no it's not necessary

**Comment by rkuhn** [comment]
LGTM (with the aforementioned restriction)

**Pull request merged**